lib/internal/bootstrap/node.js

```diff
@@ -90,7 +96,16 @@ if (isMainThread) {
 
 // Set up methods on the process object for all threads
 {
-  process.cwd = rawMethods.cwd;
+  process.cwd = () => {
+    // Cache the working directory to prevent lots of lookups. Only check this
+    // once. If the working directory is changed by `chdir`, just return the new
+    // cwd instead.
+    cwd = rawMethods.cwd();
+    process.cwd = () => {
+      return cwd;
+    };
+    return cwd;
+  };
   process.dlopen = rawMethods.dlopen;
   process.uptime = rawMethods.uptime;
 
```

Comment:
I like the idea but not the implementation. I think you can do this without overwriting the `process.cwd()` using a var to track if `cwd` changed or not then getting the new value if changed and `cwd()` is called.

Reply:
I changed it that it does not override the function anymore.